src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_utils.hpp

```diff
@@ -14,28 +14,31 @@ namespace ov {
 namespace intel_cpu {
 namespace brgemm_utils {
 
-enum class BRGEMM_TYPE {
-    STAND_ALONE,            // No extra requirements, used for f32|f32
-    WITH_AMX,               // i8|i8 or bf16|bf16 on AMX system - needs BrgemmCopyB and scratchpad
-    WITH_COMPENSATIONS,     // i8|i8 (non-AMX system) - needs BrgemmCopyB for data repacking and compensations
-    REPACKING_ONLY          // u8|i8 or bf16|bf16 (non-AMX system) - needs BrgemmCopyB on second input for data repacking
-};
-
-dnnl::impl::cpu::x64::cpu_isa_t get_primitive_isa(const ov::element::Type& dt_in0, bool is_with_amx);
-
-BRGEMM_TYPE get_brgemm_type(const element::Type& element_type_a, const Dimension& K_dim, bool transpose_b);
-
-inline bool stand_alone(BRGEMM_TYPE type) { return type == BRGEMM_TYPE::STAND_ALONE; }
-
-inline bool with_amx(BRGEMM_TYPE type) { return type == BRGEMM_TYPE::WITH_AMX; }
+class BrgemmConfig {
```

Comment:
I believe that to have only one const config in `Brgemm` and `BrgemmCopyB` is more convenient that work with static functions and different passed arguments.